presentation/src/main/java/me/androidbox/presentation/login/viewmodel/LoginViewModel.kt

```diff
@@ -4,11 +4,23 @@ import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import me.androidbox.domain.authentication.NetworkResponseState
+import me.androidbox.domain.authentication.model.LoginModel
+import me.androidbox.domain.authentication.usecase.LoginUseCase
 import javax.inject.Inject
 
 @HiltViewModel
-class LoginViewModel @Inject constructor() : ViewModel() {
+class LoginViewModel @Inject constructor(
+    private val loginUseCase: LoginUseCase
```

Comment:
You are missing the validation's usecases
Will that be on another PR?

Reply:
@MKiperszmid That will be added in a separate PR. As I want to keep the PRs shorts. So validation will have a PR on is own as its just a separate unit of work.